.github/workflows/release.yml

```diff
@@ -5,131 +5,124 @@ on:
-          sed -i ${replace} CHANGELOG.md
+          # Update Changelog
+          sed -i "s/\[unreleased\]/\[${VERSION}\]/" CHANGELOG.md
+          sed -i "s/...HEAD/\...v${VERSION}/" CHANGELOG.md
+
+      #
+      # Uses the enforcer plugin to ensure no -SNAPSHOT version are used
+      #
+      - name: Enforce no SNAPSHOT used
+        run: |
+          mvn -P no-snapshots enforcer:enforce
+
+      #
+      # Publish release branch
+      #
+      - name: Publish Release Branch
+        run: |
           # commit & push
           git add -A
           git commit -m "release ${VERSION}: updated version to ${VERSION}"
           git push -u origin release/v${VERSION}
-      # wait for status of commit to change from pending
-      - name: Wait for ci pipeline
+
+      #
+      # Wait for status of commit to change from pending
+      #
+      - name: Wait for CI pipeline
         run: |
           STATUS="pending"
           # Get commit last commit of release branch
           COMMIT=$(git rev-parse HEAD)
           echo "Listen for commit $COMMIT"
           WAITED="0"
           # Time between calls
-          SLEEP_TIME="60"
+          SLEEP_TIME="15"
           while [ "$STATUS" == "pending" ] && [ "$WAITED" -le 1800 ]
           do
-           sleep ${SLEEP_TIME}
-           WAITED=$((WAITED+SLEEP_TIME))
-           STATUS=$(gh api /repos/${{ github.repository  }}/commits/"${COMMIT}"/status -q .state)
-          echo "status : $STATUS"
-          echo "waited  $WAITED s"
+            sleep ${SLEEP_TIME}
+            WAITED=$((WAITED+SLEEP_TIME))
+            STATUS=$(gh api /repos/${{ github.repository  }}/commits/"${COMMIT}"/status -q .state)
+            echo "status : $STATUS"
+            echo "waited  $WAITED s"
           done
           echo "status : $STATUS"
           if [ "$STATUS" != "success" ]
-           then exit 1
+            then exit 1
           fi
         env:
           GITHUB_TOKEN: ${{ secrets.BOT_ACCESS_TOKEN }}
 
-      - name: Merge into Main
+      #
+      # In case the CI build was successful - we can merge everything back into the master branch
+      #
+      - name: Merge into Master
         run: |
           git checkout master
           git pull
           git merge --no-ff release/v${VERSION}
-          git tag  -a  -m "v${VERSION}" "v${VERSION}"
+          git tag -a -m "v${VERSION}" "v${VERSION}"
           git push --follow-tags
 
-      - name: Create Release
+      #
+      # Create the actual github release for the version using the actual changelog
+      #
+      - name: Create Github Release
         run: |
-          gh release create "v${VERSION}" -t "v${VERSION}"  -F CHANGELOG.md -R ${{ github.repository  }} --target master
+          gh release create "v${VERSION}" -t "v${VERSION}" -F CHANGELOG.md -R ${{ github.repository  }} --target master
         env:
           GITHUB_TOKEN: ${{ secrets.BOT_ACCESS_TOKEN }}
 
-      - name: Merge into dev
+      - name: Prepare next dev version
```

Comment:
In this step, the SDK is updated to the next snapshot version. I'm not sure if this is needed since there may not be any changes in the SDK, so we should maybe leave it up to the developers to update to the snapshot version.